src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -8698,221 +8708,1188 @@ private void VerifyParameter(CSharpCompilation comp, int index, string expectedM
-                Diagnostic(ErrorCode.ERR_NameNotInContext, "parameter").WithArguments("parameter").WithLocation(8, 24),
-                // (11,17): error CS0103: The name 'parameter' does not exist in the current context
-                //     void M2([My(parameter)] int parameter) => throw null;
-                Diagnostic(ErrorCode.ERR_NameNotInContext, "parameter").WithArguments("parameter").WithLocation(11, 17)
+                // (4,16): error CS0103: The name 'parameter' does not exist in the current context
+                //     [My(nameof(parameter))]
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "parameter").WithArguments("parameter").WithLocation(4, 16)
                 );
 
             VerifyParameter(comp, 0, null);
-            VerifyParameter(comp, 1, null);
+
+            comp = CreateCompilation(source, parseOptions: TestOptions.RegularNext);
+            comp.VerifyDiagnostics();
+
+            VerifyParameter(comp, 0, "System.Int32 C.this[System.Int32 parameter] { set; }");
         }
 
         [Fact]
-        public void ParameterScope_NotInParameterAttributeNameOf()
+        public void ParameterScope_InIndexerGetterAttributeNameOf()
         {
-            var comp = CreateCompilation(@"
+            var source = @"
 class C
 {
-    void M()
+    int this[int parameter]
     {
-        local(0);
-
-        void local([My(nameof(parameter))] int parameter) => throw null;
+        [My(nameof(parameter))]
+        get => throw null;
     }
-
-    void M2([My(nameof(parameter))] int parameter) => throw null;
 }
 
 public class MyAttribute : System.Attribute
 {
     public MyAttribute(string name1) { }
 }
-");
+";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular10);
             comp.VerifyDiagnostics(
-                // (8,31): error CS0103: The name 'parameter' does not exist in the current context
-                //         void local([My(nameof(parameter))] int parameter) => throw null;
-                Diagnostic(ErrorCode.ERR_NameNotInContext, "parameter").WithArguments("parameter").WithLocation(8, 31),
-                // (11,24): error CS0103: The name 'parameter' does not exist in the current context
-                //     void M2([My(nameof(parameter))] int parameter) => throw null;
-                Diagnostic(ErrorCode.ERR_NameNotInContext, "parameter").WithArguments("parameter").WithLocation(11, 24)
+                // (6,20): error CS0103: The name 'parameter' does not exist in the current context
+                //         [My(nameof(parameter))]
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "parameter").WithArguments("parameter").WithLocation(6, 20)
                 );
 
             VerifyParameter(comp, 0, null);
-            VerifyParameter(comp, 1, null);
+
+            comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+
+            VerifyParameter(comp, 0, "System.Int32 C.this[System.Int32 parameter].get");
```

Comment:
:memo: I don't have a strong opinion on whether the indexer parameter or the accessor parameter should be found. I went with accessor parameter because that seems more consistent with methods and other targets.